### ISSUE TITLE
make progress display with long filenames sort on full filenames not truncated ones

### ISSUE THREAD

**Issue by mlc** [rclone/rclone#4271]
#### What is the problem you are having with rclone?

When using `rclone sync --progress`, the list of currently-being-transferred files can be in a somewhat confusing order:

![image](https://user-images.githubusercontent.com/1211/82741929-be489500-9d25-11ea-965c-75092808b8bb.png)

At this moment, I'd expect to see `vol8`, `vol80`, `vol800`, and `vol801` in that order, which is the order that you get if you sort these filenames in ASCII sort order, and the order in which the transfers appear to be started.

I believe the issue is that the filenames are being truncated _before_ being sorted, and because the filenames are of different lengths, they get truncated at different points, and thus get sorted based on the essentially arbitrary character where the truncation happens.

#### What is your rclone version (output from `rclone version`)

```
$ rclone version
rclone v1.50.1-DEV
- os/arch: linux/amd64
- go version: go1.12.10
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Ubuntu, 64bit

####  Which cloud storage system are you using? (eg Google Drive)

Amazon S3

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone --ignore-checksum --progress --s3-storage-class STANDARD_IA --s3-region eu-north-1 sync`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

I didn't produce one but can try if it'll really solve some problem.

**Comment by animosity22** [comment]
I doubt you'll see a difference, but can you reproduce on the latest version rather than an old dev version?

**Comment by mlc** [comment]
Thanks for the quick response!

I upgraded to
```
rclone v1.51.0
- os/arch: linux/amd64
- go version: go1.13.7
```
and can confirm that I see the same behavior (and, for what it's worth, I also see the behavior on the `Checking:` list of files too, when that is being shown).

**Comment by ncw** [comment]
I think you are right.

https://github.com/rclone/rclone/blob/9db8ecbc3220ffa8bc956113e513f482989d0597/fs/accounting/stringset.go#L68-L96

It would be relatively easy to change it if you wanted to have a go @mlc ?

**Comment by albertony** [comment]
I'm pretty sure this was fixed with PR #4347, which was merged back in july 2020.

Closing this issue. @mlc : Please retest latest version, if there is still issues we can re-open.

**Comment by mlc** [comment]
@albertony Yes, I agree this is fixed, and has been for some time. Thanks!
